File: src/main/java/club/deneb/client/utils/clazz/Button.java
package club.deneb.client.utils.clazz;


import java.util.ArrayList;
import java.util.List;

public class Button {
    public List<VoidContainer> binds = new ArrayList<>();

    public void runBinds(){
        binds.forEach(VoidContainer::invoke);
    }

    public Button setBind(VoidContainer voidContainer){
        binds.add(voidContainer);
        return this;
    }
}

File: src/main/java/club/deneb/client/utils/clazz/Button.kt
package club.deneb.client.utils.clazz

import java.util.*
import java.util.function.Consumer

class Button {

    private var binds: MutableList<VoidContainer> = ArrayList()
    fun runBinds() {
        binds.forEach(Consumer { obj: VoidContainer -> obj.invoke() })
    }

    fun setBind(voidContainer: VoidContainer): Button {
        binds.add(voidContainer)
        return this
    }
}